### PR TITLE
Add test for Yatai service docker image

### PR DESCRIPTION
**Pull request by yubozhao**
* add test for yatai service docker image
* linter

### CODE REVIEW

File: e2e_tests/conftest.py [hunk 1]
```diff
@@ -132,32 +133,56 @@ def temporary_s3_bucket():
...
+        temp_docker_file_path = os.path.join(temp_dir, 'Dockerfile')
+        temp_docker_file = open(temp_docker_file_path, 'w')
+        temp_docker_file.write(
+            f'FROM bentoml/yatai-service:{PREV_PYPI_RELEASE_VERSION}\n'
+        )
+        temp_docker_file.write('ADD . /bentoml-local-repo\n')
+        temp_docker_file.write('RUN pip install /bentoml-local-repo\n')
+        temp_docker_file.close()
+        logger.info('building docker image')
+        docker_client.images.build(
```
Comment (parano) [inline e2e_tests/conftest.py:153]:
```
dockerfile = f"""\
FROM bentoml/yatai-service:{PREV_PYPI_RELEASE_VERSION}
ADD . /bentoml-local-repo
RUN pip install /bentoml-local-repo
"""
docker_client.images.build(
            path=local_bentoml_repo_path,
            fileobj=BytesIO(dockerfile.encode('utf-8')),
            tag=docker_tag,
)

File: e2e_tests/conftest.py [hunk 2]
```diff
@@ -132,32 +133,56 @@ def temporary_s3_bucket():
...
-
-    docker_proc = subprocess.Popen(
-        command, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-    )
-    wait_for_docker_container_ready(
-        container_name, b'* Starting BentoML YataiService gRPC Server'
-    )
+    docker_client = docker.from_env()
+    local_bentoml_repo_path = os.path.abspath(os.path.join(__file__, '..', '..'))
+    docker_tag = f'bentoml/yatai-service:local-bentoml-{uuid.uuid4().hex[:6]}'
```
Comment (parano) [inline e2e_tests/conftest.py:138]: nit: `bentoml/yatai-service:e2e-test-`

File: e2e_tests/conftest.py [hunk 3]
```diff
@@ -132,32 +133,56 @@ def temporary_s3_bucket():
...
+
+    # Note: When set both `custom_context` and `fileobj`, docker api will not use the
+    #       `path` provide... docker/api/build.py L138. The solution is create an actual
+    #       Dockerfile along with path, instead of fileobj and custom_context.
+    with TempDirectory() as temp_dir:
+        temp_docker_file_path = os.path.join(temp_dir, 'Dockerfile')
+        temp_docker_file = open(temp_docker_file_path, 'w')
```
Comment (parano) [inline e2e_tests/conftest.py:145]:
```
with open(temp_docker_file_path, 'w') as f:
   f.write(f"""\
FROM bentoml/yatai-service:{PREV_PYPI_RELEASE_VERSION}
ADD . /bentoml-local-repo
RUN pip install /bentoml-local-repo
   """
   )
)
```

File: e2e_tests/conftest.py (hunk 1 above)
Comment (yubozhao) [reply e2e_tests/conftest.py:153]: discussed offline. see notes

**Pull request merged**